src/ILCompiler.ReadyToRun/src/ObjectWriter/R2RPEBuilder.cs

```diff
@@ -301,8 +381,109 @@ public void Write(Stream outputStream)
                 CorHeaderFileOffset,
                 outputStream);
 
+            // On Linux, we must patch the section headers. This is
+            // because the CoreCLR runtime on Linux requires the 12-16 low-order
+            // bits of section RVAs (the number of bits corresponds to the page
```

Comment:
Also, should we just do this on Linux?

Reply:
Well, my thinking is that it should never hurt to patch the headers and it's better to minimize the code that runs only on one OS.